src/history.ts

```diff
@@ -70,20 +88,62 @@ export class SceneHistory {
       }, [] as Mutable<typeof elements>),
     });
 
+  shouldCreateEntry(parsedEntry: ParsedEntry): boolean {
+    if (!this.lastEntry) {
+      return true;
+    }
+    const { appState, elements } = parsedEntry;
+    // note: this is safe because ParsedEntry is guaranteed no excess props
+    let key: keyof typeof appState;
+    for (key in appState) {
+      if (key === "selectedElementIds") {
+        continue;
+      }
+      if (appState[key] !== this.lastEntry.appState[key]) {
+        return true;
+      }
+    }
+
+    for (const i in elements) {
+      const prev = elements[i];
+      const next = this.lastEntry.elements[i];
+      if (
+        !prev ||
+        !next ||
+        prev.id !== next.id ||
+        prev.version !== next.version
```

Comment:
do we need to check versionNonce here

Reply:
Maybe. I don't recall when exactly we're changing it (you certainly know more since you've introduced it) --- would have to review if that wouldn't result in undesired history entries. Shouldn't `version` be enough since we only care about actual changes to props?

Reply:
Checking code, we're setting it in tandem with `version` --- so what's the point of `versionNonce` again?

Reply:
it's possible for an element's props to change but have the version number remain the same in the event that two or more peers are editing an element concurrently and issue conflicting edits. `versionNonce` was introduced to distinguish between the two conflicting edits.

Reply:
I see. Shouldn't we make `version` use `randomInteger()` to begin with, and get rid of `versionNonce`? Though I can see we're using `>` on inbound sync to determine whether to discard remote change, but that seems brittle. But probably leave that decision for later.

I'll make the update to the history resolver to check for `versionNonce`, tomorrow.

Reply:
we need it to be an increasing integer to avoid resyncing old changes in multiplayer mode, randomInteger() won't work for that use case.